File: boj20055.cpp
#include <iostream>
#include <vector>
using namespace std;

int N, K, answer;
int A[201];
bool Belt[101];

void rotate(){
    for(int i=N-1; i>0; i--){
        if(Belt[i]){
            Belt[i+1]=true;
            Belt[i]=false;
        }
    }
    Belt[N]=false;

    int temp=A[2*N];
    for(int i=2*N; i>1; i--){
        A[i]=A[i-1];
    }
    A[1]=temp;
}

void robot(){
    for(int i=N-1; i>0; i--){
        if(Belt[i] && !Belt[i+1] && A[i+1]>0){
            Belt[i+1]=true;
            Belt[i]=false;
            A[i+1]--;
        }
    }
    Belt[N]=false;
}

int count(){
    int cnt=0;
    for (int i=1; i<=2*N; i++){
        if(A[i]==0){
            cnt++;
        }
    }
    return cnt;
}

int main(){
    cin>>N>>K;
    for (int i =1; i<=2*N; i++){
        cin>>A[i];
    }
    answer=0;
    while(true){
        rotate();
        robot();
        if(!Belt[1] && A[1]>0){
            Belt[1]=true;
            A[1]--;
        }
        answer++;
        if (count()>=K){
            break;
        }
    }
    cout<<answer;
	system("pause");
    return 0;
}

File: boj20055.py
from collections import deque
import sys

input=sys.stdin.readline
N, K=map(int, input().split())
A=deque(map(int, input().split()))
answer=0
robot=deque([0]*(N*2))

while(True):
    A.rotate(1)
    robot.rotate(1)
    robot[N-1]=0

    for i in range(N-2, -1, -1):
        if(robot[i]!=0 and robot[i+1]==0 and A[i+1]>=1):
            A[i+1]-=1
            robot[i+1]=robot[i]
            robot[i]=0
    robot[N-1]=0

    if(robot[0]==0 and A[0]>0):
        A[0]-=1
        robot[0]=1
    
    cnt=0
    for i in range(N*2):
        if(A[i]==0):
            cnt+=1
    answer+=1
    if(cnt>=K):
        print(answer)
        break
    